apps/condo/domains/common/components/AudioPlayer.tsx

```diff
@@ -0,0 +1,155 @@
+
+const { Option } = Select
+
+const StyledSelect = styled(Select)`
+    .ant-select-selector {
+      border: none !important;
+      box-shadow: none !important;
+      
+      .ant-select-selection-item {
+        font-size: 14px;
+      }
+    }
+`
+
+const PLAYER_ROW_GUTTER: [Gutter, Gutter] = [40, 0]
+const RECORD_ROW_GUTTER: [Gutter, Gutter] = [12, 0]
+const SPEED_ROW_GUTTER: [Gutter, Gutter] = [8, 0]
+const PLAYER_ICON_WRAPPER_STYLE: CSSProperties = { display: 'flex', alignItems: 'center', justifyContent: 'center' }
+const WAVE_WRAPPER_STYLE: CSSProperties = { width: '100%', height: '28px' }
+
+interface IAudioPlayerProps {
+    src: string
+    trackId: string
+    autoPlay?: boolean
+}
+
+export const AudioPlayer: React.FC<IAudioPlayerProps> = ({ trackId, src, autoPlay }) => {
+    const intl = useIntl()
+    const SpeedMessage = intl.formatMessage({ id: 'ticket.callRecord.speed' })
+
+    const [playing, setPlaying] = useState(false)
+    const [currentTime, setCurrentTime] = useState('00:00')
+    const [totalTime, setTotalTime] = useState('00:00')
+    const [speed, setSpeed] = useState(1)
+    const waveformRef = useRef<HTMLDivElement>(null)
+    const waveform = useRef<WaveSurfer>()
+
+    useEffect(() => {
+        if (typeof window !== 'undefined') {
+            const track: HTMLMediaElement = document.querySelector(`#${trackId}`)
+
+            waveform.current = WaveSurfer.create({
+                container: waveformRef.current,
+                height: 28,
+                progressColor: createGradient(),
+                waveColor: colors.gray[5],
+                responsive: true,
+                barGap: 1,
+                barWidth: 2,
+                barHeight: 28,
+                cursorWidth: 0,
+                hideScrollbar: true,
+            })
+
+            waveform.current.on('ready', () => {
+                setTotalTime(formatTime(waveform.current.getDuration()))
+
+                if (autoPlay) {
+                    waveform.current.play()
+                    setPlaying(true)
+                }
+            })
+
+            waveform.current.on('audioprocess', () => {
+                setCurrentTime(formatTime(waveform.current.getCurrentTime()))
+            })
+
+            waveform.current.on('finish', () => {
+                setPlaying(false)
+            })
+
+            waveform.current.load(track)
+
+            return () => {
+                if (waveform.current) {
+                    waveform.current.destroy()
+                }
+            }
+        }
+    }, [])
+
+    const handlePlay = () => {
+        setPlaying(!playing)
+        waveform.current.playPause()
+    }
+
+    const handleSpeedChange = (value: number) => {
+        setSpeed(value)
+        waveform.current.setPlaybackRate(value)
+    }
+
+    const PlayerIcon = playing ? Pause : Play
+
+    const formatTime = (time: number): string => {
+        const minutes = Math.floor(time / 60).toString().padStart(2, '0')
```

Comment:
What if it longer than hour somehow, `105:37` would be a valid value?

Reply:
made the time display as `hh:mm:ss` if total duration > 1h